drivers/sx127x/sx127x_getset.c

```diff
@@ -127,7 +127,7 @@ void sx127x_set_channel(sx127x_t *dev, uint32_t channel)
     /* Save current operating mode */
     dev->settings.channel = channel;
 
-    channel = (uint32_t)((double) channel / (double)LORA_FREQUENCY_RESOLUTION_DEFAULT);
+    channel = (uint32_t)((double) channel / (double)SX127X_FREQUENCY_STEP);
```

Comment:
```suggestion
    channel = (channel << 8) / 15625;
```

Reply:
This can produce and offset of 256 Hz in the carrier frequency. It shouldn't be a problem, but I would need to test to ensure nothing is broken.

Reply:
in general, I really think this driver should be reworked in order to remove all floating point stuff. I used an alternative function for calculating Time on Air in GNRC LoRaWAN and saved around 1 KB of ROM

Reply:
I might be a bit dense, but how do you arrive at the 256 Hz offset?
I assumed the math should work out the same.

Reply:
Ah no, you are right (I switched the numbers in the fraction).
Since it's a Frequency Step the proposed gives exactly the same value for the internal register